### ISSUE TITLE
Optitron fails to load on trivial install

### ISSUE THREAD

**Issue by cowboyd** [joshbuddy/optitron#5]
I can't seem to get it working.

Using OSX and rvm to create a pristine gem environment, optitron does not properly install and seems to have unworkable dependencies

```
legolas:~ cowboyd$ rvm use 1.8.7@optitron --create

info: Using ruby 1.8.7 p174 with gemset optitron
legolas:~ cowboyd$ gem install optitron --no-rdoc --no-ri
ERROR:  Error installing optitron:
        optitron requires sexp_processor (= 3.0.4, runtime)
legolas:~ cowboyd$ gem install optitron --no-rdoc --no-ri
Successfully installed sexp_processor-3.0.4
Successfully installed optitron-0.1.8
2 gems installed
legolas:~ cowboyd$ ruby -rubygems -e "require 'optitron'"
/Users/cowboyd/.rvm/rubies/ruby-1.8.7-p174/lib/ruby/site_ruby/1.8/rubygems.rb:233:in `activate': can't activate sexp_processor (= 3.0.4, runtime) for ["optitron-0.1.8"], already activated sexp_processor-3.0.5 for ["ruby_parser-2.0.5", "optitron-0.1.8"] (Gem::LoadError)
        from /Users/cowboyd/.rvm/rubies/ruby-1.8.7-p174/lib/ruby/site_ruby/1.8/rubygems.rb:249:in `activate'
        from /Users/cowboyd/.rvm/rubies/ruby-1.8.7-p174/lib/ruby/site_ruby/1.8/rubygems.rb:248:in `each'
        from /Users/cowboyd/.rvm/rubies/ruby-1.8.7-p174/lib/ruby/site_ruby/1.8/rubygems.rb:248:in `activate'
        from /Users/cowboyd/.rvm/rubies/ruby-1.8.7-p174/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:35:in `require'
        from -e:1
```

**Comment by joshbuddy** [comment]
Hey, I've changed the deps to be more loose, any chance you can grab the latest and test it?

**Comment by cowboyd** [comment]
that seems to fix the dependency error. I'm getting a weird error in ruby_parser now, but I'm under the gun, so I'm going to fall back to optparse for today. I look forward to giving optitron another go soon though.

**Comment by joshbuddy** [comment]
If you can give me the code to reproduce it, or even the error, I'm happy to take a look at it!

**Comment by cowboyd** [comment]
Yeah, I had a bit more time later to revisit it. I think it was having a method without a desc(). It is just a obscure error for what's really going on.

here is an example.

https://gist.github.com/675b278d88dd0bb71a14

I'll try again to convert my parser to optitron, and let you know how it goes.

**Comment by joshbuddy** [comment]
I'm not in front of my computer right now, but I would immediately suggest you move the dispatch to outside the class definition. Let me know if that works. I'll take a look too, and I could certainly improve error messaging around that

**Comment by cowboyd** [comment]
actually, dispatch inside the class works. It turns out that my other problem was that it assumes that class names are not nested.

```
class Foo::Bar < Optitron::CLI
```

is problematic, but 

```
FooBar < Optitron::CLI
```

is no problem. In any case, It's working well for me now. Thanks!

**Comment by joshbuddy** [comment]
Oh wow, nice find. I'll fix the nesting issue. Thanks for all your help!

**Comment by joshbuddy** [comment]
Fixed the nesting issue in 0.1.9
